packages/tempus/src/Tempus_AuxModel.hpp

```diff
@@ -0,0 +1,40 @@
+// @HEADER
+// ****************************************************************************
+//                Tempus: Copyright (2017) Sandia Corporation
+//
+// Distributed under BSD 3-clause license (See accompanying file Copyright.txt)
+// ****************************************************************************
+// @HEADER
+
+#ifndef Tempus_AuxModel_hpp
+#define Tempus_AuxModel_hpp
+
+
+#include "Tempus_ODEParameters.hpp"
+
+
+namespace Tempus {
+
+
+/** \brief A model to compute auxiliary variables.
+ *
+ *  This could grow into the Tempus version of ModelEvaluator.
+ */
+template <typename Scalar>
+class AuxModel
+{
+public:
+
+  virtual void evaluateAuxModel(
+  const Teuchos::RCP<Thyra::VectorBase<Scalar> > & y,
```

Comment:
Will AuxModel implementations be allowed to keep RCPs to any of these input arguments?  If not, then you should pass the arguments by const reference (if they are always nonnull) or raw pointer (if they may be nonnull) or `Teuchos::Ptr` (if they may be nonnull).

Passing an `RCP<T>` (or `const RCP<T>&`) into an instance of a class suggests that you intend to share ownership.  I'm guessing that's not what you intend here.  Implementations of this method will likely just use the objects during the method and not keep RCPs to them.

Reply:
Hi Mark,  thanks for the "pointers" as I am not very savvy about these things. :)

But I has following the ModelEvaluator::evalModelImpl() example on this.  Is there a reason it is done there?

Reply:
I'm kind of intruding here -- I was just looking through GitHub notifications while waiting on a long build.  I'm not sure about the history of ModelEvaluator.  If you don't need interface compatibility with ModelEvaluator, then I think you should feel free to do the right thing.  If you _do_ need interface compatibility with ModelEvaluator, then you're stuck, I guess.